cmd/cspc-operator/app/cleanup.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	bdc "github.com/openebs/maya/pkg/blockdeviceclaim/v1alpha1"
+	apiscspc "github.com/openebs/maya/pkg/cstor/poolcluster/v1alpha1"
+	cspi "github.com/openebs/maya/pkg/cstor/poolinstance/v1alpha3"
+	util "github.com/openebs/maya/pkg/util"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func cleanupCSPIResources(cspcObj *apis.CStorPoolCluster) error {
+	cspiList, err := cspi.NewKubeClient().WithNamespace(cspcObj.Namespace).List(
+		metav1.ListOptions{
+			LabelSelector: string(apis.CStorPoolClusterCPK) + "=" + cspcObj.Name,
+		},
+	)
+	if err != nil {
+		return err
+	}
+	opts := []cspiCleanupOptions{cleanupBDC}
+	for _, cspiObj := range cspiList.Items {
+		cspiObj := cspiObj // pin it
+		if cspiObj.DeletionTimestamp != nil && len(cspiObj.Finalizers) == 1 {
```

Comment:
We want to enter this block when we have only `CSPC finalizer` present on CSPC. Being more specific on what we want will ensure the code is extensible.
IMO, we should not care about the length of finalizer array but the actual finalizer which the operator is supposed to remove. 
```
 if cspiObj.DeletionTimestamp != nil &&HasCSPCFinalizer(cspiObj)== 1{
...
}

func HasCSPCFinalizer(cspiObj){
// Logic to decide wether the finalizer is present 
}
```

Reply:
Done